### PR TITLE
Allow username in login

### PR DESCRIPTION
**Pull request by microArtur**
https://github.com/nanoPayinc/NANOPAY/pull/15024

https://nanopay.atlassian.net/browse/NP-7398

### CODE REVIEW

**Comment by microArtur** [comment]
https://nanopay.atlassian.net/browse/NP-7398

**Review by Nauna — APPROVED** [review]

File: src/foam/nanos/u2/navigation/SignIn.js [hunk 1]
```diff
@@ -173,6 +186,9 @@ foam.CLASS({
               await this.nextStep();
             }
           } catch (err) {
+              if ( this.DuplicateEmailException.isInstance(err.data.exception) ) {
```
Comment (Nauna) [inline src/foam/nanos/u2/navigation/SignIn.js:193]: so you added the exception you wanted for this path... but this error is also currently thrown in foam/nanos/auth/RetrievePassword.js
foam/nanos/auth/ruler/PreventDuplicateEmailAction.js
nanopay/auth/NanopayResetPasswordTokenService.js

can you guarantee the correct flow by just checking the type/instance ? Perhaps also checking the message...

File: src/foam/nanos/u2/navigation/SignIn.js [hunk 2]
```diff
@@ -46,6 +47,18 @@ foam.CLASS({
       hidden: true,
       transient: true
     },
+    {
+      class: 'String',
+      name: 'username',
```
Comment (jlhughes) [inline src/foam/nanos/u2/navigation/SignIn.js:52]: Why is this required? Why does identifier not work?

Comment (microArtur) [reply src/foam/nanos/u2/navigation/SignIn.js:52]: We want to ask user to enter username in case of Duplicate email found on the system.
https://nanopay.atlassian.net/browse/NP-7398

Comment (jlhughes) [reply src/foam/nanos/u2/navigation/SignIn.js:52]: Changing the field label from identifer to username on second attempt would not suffice? 
Or are you then presenting three fields?  identifier, username, password?

**Review by jlhughes — APPROVED** [review]
